docs/gettinginvolved/plugins/building-driver.rst

```diff
@@ -0,0 +1,161 @@
+Passing all tests ultimately results in certification of your driver to work with Flocker.
+   
+Before beginning the testing steps below, you can review our :ref:`build-flocker-driver-faq` for common issues encountered during driver development and testing.
+
+Testing your Driver
+===================
+
+#. Implement minimal functional tests.
+
+   To test that your implementation is correct you should instantiate a generic test suite that makes sure your class correctly implements the interface:
+
+   .. code-block:: python
+
+      from uuid import uuid4
+      from flocker.node.agents.test.test_blockdevice import make_istatechange_tests
+
+      def api_factory(test):
+          # Return an instance of your IBlockDeviceAPI implementation class, given
+          # a twisted.trial.unittest.TestCase instance.
+
+      # Smallest volume to create in tests, e.g. 1GiB:
+      MIN_ALLOCATION_SIZE = 1024 * 1024 * 1024
+
+      # Minimal unit of volume allocation, e.g. 1MiB:
+      MIN_ALLOCATION_UNIT = 1024 * 1024
+
+      class YourStorageTests(make_istatechange_tests(
+          api_factory, MIN_ALLOCATION_SIZE, MIN_ALLOCATION_UNIT,
+          # Factory for valid but unknown volume id specific to your backend:
+          lambda test: unicode(uuid4()))):
+          """
+          Tests for your storage.
+          """
+
+   If you wish the tests to cleanup volumes after each run, please provide a cleanup version of ``IBlockDeviceAPI``.
+   For an example of a clean up script, see the `EBS API with cleanup <https://github.com/ClusterHQ/flocker/blob/master/flocker/node/agents/test/blockdevicefactory.py#L225>`_ inside ``api_factory``.
+
+   You can run these tests with the ``trial`` test runner, provided by `Twisted <https://twistedmatrix.com/trac/>`_, one of Flocker's dependencies:
+
+   .. prompt:: bash $
+
+      trial yourstorage.test_yourstorage
+
+#. Additional functional tests.
+
+   You are encouraged to write additional functional tests to cover logic specific to your driver implementation.
+   For example, here are some `EBS driver-specific tests <https://github.com/ClusterHQ/flocker/blob/master/flocker/node/agents/functional/test_ebs.py#L155>`_ written by ClusterHQ.
+
+#. Run acceptance tests.
+
+   After all functional tests pass, please run acceptance tests according to our :ref:`acceptance-testing` documentation.
+
+#. Setup Continuous Integration environment for tests.
+
+   After your acceptance tests run clean, please set up a CI environment for functional and acceptance tests for your driver.
+   For example:
+   
+   * `EBS functional tests <http://build.clusterhq.com/builders/flocker%2Ffunctional%2Faws%2Fubuntu-14.04%2Fstorage-driver>`_
+   * `EBS acceptance tests <http://build.clusterhq.com/builders/flocker%2Facceptance%2Faws%2Fubuntu-14.04%2Faws>`_
+
+
+Publishing your driver
+======================
+
+Once CI test runs pass for a week, you are ready to publish your driver and assert that is certified to work with Flocker.  
```

Comment:
It is difficult to determine what is a ClusterHQ recommendation and what is necessary. Does this mean that I can't assert that my driver certified to work with Flocker unless I've been running it unchanged for 1 week on a CI environment? Must it be public and on GitHub? (not BitBucket?) to do this? Must I include an Apache 2.0 License, or is that recommended?